plugin-api/beta/client/hocs/withTags.js

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {compose, gql} from 'react-apollo';
+import {getDisplayName} from 'coral-framework/helpers/hoc';
+import {capitalize} from 'coral-framework/helpers/strings';
+import withMutation from 'coral-framework/hocs/withMutation';
+import withFragments from 'coral-framework/hocs/withFragments';
+import {showSignInDialog} from 'coral-framework/actions/auth';
+import {addNotification} from 'coral-framework/actions/notification';
+
+export default (tag) => (WrappedComponent) => {
+  if (typeof tag !== 'string') {
+    console.error('Tag must be a valid string');
+    return null;
+  }
+
+  tag = tag.toLowerCase();
+  const Tag = capitalize(tag);
+
+  const COMMENT_FRAGMENT = gql`
+    fragment Coral_UpdateFragment on Comment {
+      tags {
+        tag {
+          name
+        }
+      }
+    }
+  `;
+  
+  const isTagged = (tags) =>
+  !!tags.filter((t) => t.tag.name === Tag.toUpperCase()).length;
+  
+  const withAddTag = withMutation(
```

Comment:
The mutation should not be defined here, but a more generic version should exist in framework.